django/db/backends/postgresql/base.py

```diff
@@ -257,6 +258,16 @@ def init_connection_state(self):
             if not self.get_autocommit():
                 self.connection.commit()
 
+        # Set the role on the connection. This is useful if the
+        # credential used to login is not the same as the role that
+        # owns database resources. This scenario can occur when using
+        # temporary credentials generated by an external system like
+        # Vault.
+        options = self.settings_dict["OPTIONS"]
+        if options.get("assume_role"):
+            with self.connection.cursor() as cursor:
+                cursor.execute("SET ROLE %s", (options["assume_role"],))
```

Comment:
Shouldn't this also utilize:
```
            if not self.get_autocommit():
                self.connection.commit()
```

like the timezone stuff above? I think the same argument holds true there.

Reply:
Ah yeah, it looks like this is documented behavior of [SET](https://www.postgresql.org/docs/8.4/sql-set.html)

> If SET (or equivalently SET SESSION) is issued within a transaction that is later aborted, the effects of the SET command disappear when the transaction is rolled back. Once the surrounding transaction is committed, the effects will persist until the end of the session, unless overridden by another SET.

Fixed.